site/download.md

```diff
@@ -65,14 +54,18 @@ Client guide | API docs
 [The Pulsar C++ client](../docs/latest/clients/Cpp) | [Doxygen docs](../api/cpp)
 
 
+<!--
```

Comment:
Is this intended to be commented out?

Reply:
@merlimat For now, yes, because older releases aren't currently available here: http://www.apache.org/dist/incubator/pulsar/

Reply:
As pointed out, we shouldn't use `http://www.apache.org/dist/incubator/pulsar/` but rather, for older releases, point to the "archive" URL: `http://archive.apache.org/dist/incubator/pulsar/pulsar...` as it was before